### ISSUE TITLE
Manual replaceOnChange: ["appId"] required for changed amplify app id on associated resources

### ISSUE THREAD

**Issue by Jimmy89** [pulumi/pulumi-aws-native#822]
### What happened?

During an update we changed the appId from an existing amplify app. Pulumi had to "update" the branch associating with the old appId to the new one. Instead of re-creating the amplify branch, it tried to update the appId of the branch, which logically generated an error by AWS. By default the aws-native package should know that when an appId changes of an amplify resource that the associated resource must be replaced (instead of updating the property).

### Expected Behavior

Replace when appId changes, not update.

### Steps to reproduce

```
const amplifyBranch = new aws.amplify.Branch(`branch`, {
  appId: amplifyApp.id,
  branchName,
  enableAutoBuild: true,
  enablePullRequestPreview: false,
  stage: "PRODUCTION",
}); // This gives an error when amplifyApp.id changed.
//}, {replaceOnChanges: ['appId']}); // This code is required instead to work correctly.

```

Error:
```
  aws-native:amplify:Branch (branch):
    error: operation error CloudControl: UpdateResource, https response error StatusCode: 400, RequestID: XXXXX, NotUpdatableException: Invalid patch update: createOnlyProperties [/properties/AppId] cannot be updated
```

### Output of `pulumi about`

```
Dependencies:
NAME                VERSION
@pulumi/aws-native  0.51.0
@pulumi/pulumi      3.55.0
@pulumi/random      4.11.2
```

### Additional context

I did not check, but maybe `aws.amplify.Domain` should be checked on this behavior as well.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Jimmy89** [comment]
This should be added to the list of https://github.com/pulumi/pulumi-aws-native/issues/736

**Comment by danielrbradley** [comment]
Hi @Jimmy89 thanks for logging this. You're correct this is the same root cause of #736 and will be solved once the core issue is addressed. I've updated the tracking list in #736 with this issue.

**Comment by mjeffryes** [comment]
This should be resolved by #1053; changes to appId will result in a replace rather than an update.
